src/skdh/completeness/utils.py

```diff
@@ -0,0 +1,508 @@
+import re
+import warnings
+import copy
+import os
+import pandas as pd
+import numpy as np
+from skdh.completeness.helpers import from_unix
+
+def compute_summary_metrics(completeness_master_dic, time_periods, timescales, measures):
+    dic_summary = {}
+    for period in time_periods:
+        period_summary = []
+
+        # Charging
+        if completeness_master_dic['charging']['all'] is None:
+            charging_time = None
+        else:
+            charging_time = np.sum(completeness_master_dic['charging'][period][:, 1] -
+                                   completeness_master_dic['charging'][period][:, 0])
+        period_summary.append(['charge time', charging_time])
+
+        # Wearing
+        if completeness_master_dic['wearing']['all'] is None:
+            wear_time = None
+        else:
+            wear_time = np.sum(completeness_master_dic['wearing'][period]['wear_times'][:, 1] -
+                               completeness_master_dic['wearing'][period]['wear_times'][:, 0])
+            period_summary.append(['wear time', wear_time])
+            no_wear_time = np.sum(completeness_master_dic['wearing'][period]['no_wear_times'][:, 1] -
+                               completeness_master_dic['wearing'][period]['no_wear_times'][:, 0])
+            period_summary.append(['no-wear time', no_wear_time])
+            unknown_wear_time = np.sum(completeness_master_dic['wearing'][period]['unknown_times'][:, 1] -
+                               completeness_master_dic['wearing'][period]['unknown_times'][:, 0])
+            period_summary.append(['unknown wear time', unknown_wear_time])
+
+        # Measures
+        for measure in measures:
+            if not measure in list(completeness_master_dic.keys()):
+                period_summary.append([measure + ' completeness: 0 (no valid values)', 0])
+            else:
+                for key in completeness_master_dic[measure]['Completeness']['native'][period].items():
+                    period_summary.append([measure + ', ' + key[0] + ', native', key[1]])
+                if not timescales is None:
+                    for timescale in timescales:
+                        for key in completeness_master_dic[measure]['Completeness'][timescale][period].items():
+                            period_summary.append([measure + ', ' + str(key[0]) + ', ' + str(timescale), key[1]])
+                if 'data_gaps' in completeness_master_dic[measure].keys():
+                    for data_gap in completeness_master_dic[measure]['data_gaps'][period].items():
+                        for reason in data_gap[1].items():
+                            period_summary.append([measure + ', data gap ' + str(data_gap[0]) + ', reason: ' + reason[0], reason[1]])
+
+        period_summary = np.array(period_summary)
+        dic_summary.update({period : period_summary[:, 1]})
+    df_summary = pd.DataFrame(dic_summary, index=period_summary[:, 0])
+
+    return df_summary
+
+
+def dic_to_str(dic):
```

Comment:
Look into just converting to json using `json.dumps` instead

Reply:
Is there an added benefit using json as compared to csv as we're currently doing? I don't have a strong preference for either but many people might prefer csv for easier reading

Reply:
CSV is fine, this was just for converting a dictionary to a string

Reply:
Gotcha. Removed this since it wasn't used anyway